articles/embedded.md

```diff
@@ -0,0 +1,309 @@
+ROS 2 stacks and nodes running on more powerful microprocessors. For 
+a standard ROS 2 node, communication (topics, services) from and to
+software running on the embedded ROS 2 stack should be transparent. 
+The same should hold for other core concepts such as parameters, 
+graph introspection, and run-time reconfiguration by the node lifecycle.
+
+### W-PORTABILITY: *Portability of ROS 2-based software*
+
+The embedded ROS 2 stack shall resemble or directly use the ROS 2 API 
+-- more precisely the rclcpp API -- to facilitate porting standard 
+ROS 2 nodes to MCUs.
+
+### W-DEVICES *Support of a broad spectrum of device classes*
+
+The embedded ROS 2 stack shall support a broad range of MCU device 
+classes, starting from a few tens kilobyte of RAM.
+
+### W-RTOS: *Support of a broad spectrum of RTOS(s)* 
+
+The stack shall be adaptable to different RTOS (e.g., NuttX, FreeRTOS, 
+Zephyr) and provide abstractions with regard to the application 
+software to facilitate porting application components to between RTOSs.
+
+### W-COMM: *Support of prevalent communication technologies* 
+
+To cover the broad range of use-cases for MCUs in robotics, the embedded 
+ROS 2 stack shall be usable with the default ROS 2 middleware standard 
+DDS, simple (custom) serial communication protocols just as common 
+wireless technologies like 6LoWPAN
+
+### W-MODULARITY: *High Modularity* 
+
+The embedded ROS 2 stack shall be highly modular, not only to allow for 
+adaptation to specific hardware features and different RTOS, but also to
+allow the integration with existing frameworks and to be able to create 
+customized sub-stacks. For example, it shall be usable in the style of 
+rosserial, thus bringing basically the message serialization to the MCU 
+only. Similarly, it should be possible to derive a sub-stack that provides 
+node graph features and publish/subscribe but no parameters and services, 
+and so on.
+
+### W-CONTROL: *Support control-oriented applications*  
+
+MCUs are great for control, and micro-ROS should be as well. This usually 
+means hard real-time performance with low jitter and small response times.
+
+### W-POWER *Make low-power modes possible* 
+
+MCUs are often used in battery-powered applications, and/or in applications
+with a large amount of standby time. Therefore, the stack should make it 
+easily possible to save power.
+
+### W-STATIC: *Use static initialization* 
+
+Static initialization is less error-prone, easier to analyze, and shifts 
+memory usage from RAM to flash. It is a requirement for going to very small 
+resource use. 
+
+### W-BOOT: *Quick boot times compared to native ROS 2 machines* 
+
+Microcontrollers provide with the capability of booting very quickly
+(typically, in the order of tenths of milliseconds or less). This enhances
+the existing landscape of ROS 2-native devices which in the best cases, 
+require a few seconds to boot an embedded Linux.
```

Comment:
Any improvements to initialisation times for the existing ROS2 stack are going to be welcomed by anyone working at any scale.

Reply:
It is important to know where this comes from. Some of the main ingredients are:
 1. Know your hardware at compile time (so, no probing for addresses, no trying unused drivers)
 1. Compile everything statically (so, no symbol remapping at load time)
 1. No use of hardware which takes long to initialize

And then there is making sure that each node's initialization method runs fast, which is not a framework issue as such.

This is just to say: Don't expect miracle from this work on your Linux box ;-)